specification/containerservices/resource-manager/Microsoft.ContainerService/stable/2018-03-31/managedClusters.json

```diff
@@ -0,0 +1,1290 @@
+          "description": "VNet SubnetID specifies the vnet's subnet identifier. If you specify either master VNet Subnet, or agent VNet Subnet, you need to specify both. And they have to be in the same VNet."
+        },
+        "maxPods": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Maximum number of pods that can run on a node.",
+          "default": 30
+        },
+        "osType": {
+          "$ref": "#/definitions/OSType",
+          "description": "OsType to be used to specify os type. Choose from Linux and Windows. Default to Linux."
+        }
+      },
+      "required": [
+        "name",
+        "vmSize"
+      ],
+      "description": "Profile for the container service agent pool."
+    },
+    "ContainerServiceWindowsProfile": {
+      "properties": {
+        "adminUsername": {
+          "type": "string",
+          "description": "The administrator username to use for Windows VMs.",
+          "pattern": "^[a-zA-Z0-9]+([._]?[a-zA-Z0-9]+)*$"
+        },
+        "adminPassword": {
+          "type": "string",
+          "description": "The administrator password to use for Windows VMs.",
+          "pattern": "^(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%\\^&\\*\\(\\)])[a-zA-Z\\d!@#$%\\^&\\*\\(\\)]{12,123}$"
+        }
+      },
+      "required": [
+        "adminUsername",
+        "adminPassword"
+      ],
+      "description": "Profile for Windows VMs in the container service cluster."
+    },
+    "ContainerServiceLinuxProfile": {
+      "properties": {
+        "adminUsername": {
+          "type": "string",
+          "description": "The administrator username to use for Linux VMs.",
+          "pattern": "^[a-z][a-z0-9_-]*$"
+        },
+        "ssh": {
+          "$ref": "#/definitions/ContainerServiceSshConfiguration",
+          "description": "SSH configuration for Linux-based VMs running on Azure."
+        }
+      },
+      "required": [
+        "adminUsername",
+        "ssh"
+      ],
+      "description": "Profile for Linux VMs in the container service cluster."
+    },
+    "ContainerServiceNetworkProfile": {
+      "properties": {
+        "networkPlugin": {
+          "type": "string",
+          "enum": [
+            "azure",
+            "kubenet"
+          ],
+          "default": "kubenet",
+          "x-ms-enum": {
+            "name": "NetworkPlugin",
+            "modelAsString": true
+          },
+          "description": "Network plugin used for building Kubernetes network."
+        },
+        "networkPolicy": {
+          "type": "string",
+          "enum": [
+            "calico",
+            "cilium"
+          ],
+          "x-ms-enum": {
+            "name": "NetworkPolicy",
+            "modelAsString": true
+          },
+          "description": "Network policy used for building Kubernetes network."
+        },
+        "podCidr": {
+          "type": "string",
+          "pattern": "^([0-9]{1,3}\\.){3}[0-9]{1,3}(\\/([0-9]|[1-2][0-9]|3[0-2]))?$",
```

Comment:
Thanks for adding the pattern match!

Could you also add a default for pod cidr: "10.244.0.0/16" since other fields have a default value now?

Reply:
Good idea--added. I just took the previous defaults from the portal UX and this one was empty there.